ShopAppTests/App/Account/ViewModels/AddressFormViewModelSpec.swift

```diff
@@ -0,0 +1,187 @@
+                    viewModel.updateStates(with: "Country1")
+                }
+            }
+            
+            context("if states found") {
+                it("should return array of states") {
+                    countriesUseCaseMock.returnStates = true
+                    viewModel.getCountries()
+                    
+                    viewModel.namesOfStates
+                        .subscribe(onNext: { names in
+                            expect(names.isEmpty) == false
+                        })
+                        .disposed(by: disposeBag)
+                    
+                    viewModel.updateStates(with: "Country1")
+                }
+            }
+        }
+        
+        describe("when submit button did press") {
+            var disposeBag: DisposeBag!
+            var address: Address!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+                
+                address = Address()
+                address.id = "Address ID"
+                address.firstName = "First name"
+                address.lastName = "Last name"
+                address.address = "Address"
+                address.secondAddress = "Second address"
+                address.city = "City"
+                address.country = "Country"
+                address.state = "State"
+                address.zip = "Zip"
+                address.phone = "Phone"
+            }
+            
+            it("should return address with correct fields") {
+                viewModel.address = address
+                viewModel.updateFields()
+                
+                viewModel.filledAddress
+                    .subscribe(onNext: { filledAddress in
+                        expect(filledAddress.id) == address.id
+                        expect(filledAddress.firstName) == address.firstName
+                        expect(filledAddress.lastName) == address.lastName
+                        expect(filledAddress.address) == address.address
+                        expect(filledAddress.secondAddress) == address.secondAddress
+                        expect(filledAddress.city) == address.city
+                        expect(filledAddress.country) == address.country
+                        expect(filledAddress.state) == address.state
+                        expect(filledAddress.zip) == address.zip
+                        expect(filledAddress.phone) == address.phone
+                    })
+                    .disposed(by: disposeBag)
+                
+                viewModel.submitTapped.onNext()
+            }
+        }
+        
+        describe("when address valid state changed") {
+            var disposeBag: DisposeBag!
+            var address: Address!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+            }
+            
+            context("if address not filled") {
+                beforeEach {
+                    address = Address()
```

Comment:
You can name this field like `validAddress` or `invalidAddress` for more readable - not issue, but can improve tests in future